pvHelpers/misc.py

```diff
@@ -567,42 +567,40 @@ def difference(self, other):
 class CaseInsensitiveDict(dict):
     @classmethod
     def _k(cls, key):
-        return key.lower() if isinstance(key, basestring) else key
+        if isinstance(key, basestring):
+            return key.lower()
+        elif isinstance(key, tuple):
+            return tuple([cls._k(i) for i in key])
```

Comment:
do we need to sort it too?

Reply:
also, we should only support string or tuple of strings. Otherwise, it is hard to make this function correctly.

Reply:
> do we need to sort it too? and use set() if we require that each element of the tuple is unique?

what u describing is not case insensitivity

Reply:
> also, we should only support string or tuple of strings. Otherwise, it is hard to make this function correctly.

Not really. This structure is supposed to work identical to a normal dictionary except for it being case insensitive treating string based keys.

Reply:
For example, are we treating k1=("pv", "0") and k2=("0", "pv") the same key? I think the intention is we are. But this implementation will break?

Reply:
no, those are different keys. that's not the intention of case insensitivity